docs/JLBP-0022.md

```diff
@@ -0,0 +1,56 @@
+---
+jlbp:
+  id: JLBP-22
+permalink: /JLBP-22
+---
+
+# Declare all dependencies
+
+If your code references a class—for example, by invoking a method in that class—declare a dependency that includes that class in your pom.xml, build.gradle,
+or equivalent. In Bazel this practice is called "strict deps".
+
+Code should not call methods, reference fields, or instantiate classes from _indirect_ dependencies. These are dependencies of the declared dependencies. Projects that rely on indirect dependencies have an annoying habit of breaking in unexpected ways when direct dependencies are upgraded. 
+
+For example, your project might declare a dependency on the 
+Google HTTP Java Client which
+itself depends on the Apache HTTP Components. If so, it is possible to use the `org.apache.http.client.utils.URLEncodedUtils` method in your own project without
+explicitly declaring a dependency on Apache HTTP Components. However, you should
+add the dependency anyway. This way if a future version of the
+Google HTTP Java Client no longer depends on  Apache HTTP Components, your code will
+still compile and work. Strict dependencies also help static analysis tools better understand a project.
```

Comment:
I think you could start a new paragraph with "String dependencies...." It's starting a new idea.